### ISSUE TITLE
FocusZone in DetailList component doesn't support Multiline TextField

### ISSUE THREAD

**Issue by woohling** [microsoft/fluentui#12177]
### Environment Information

- **Package version(s)**: 7.88.0
- **Browser and OS versions**:  MacOS 10.14.0 and Chrome 80.0

### Please provide a reproduction of the bug in a codepen:

Fabric Doc link: https://developer.microsoft.com/en-us/fabric/#/controls/web/focuszone
Example: **Nesting FocusZones in list rows**, just click the 'export to codepen'. And change the TextFiled with `multiline={true}` property.

#### Actual behavior:
1. Focus on `<TextField multiple />` component, and provide inputs, and use arrow key to navigate
2. other cells will be navigated and focused.

#### Expected behavior:
1. When using the arrow key in TextField component, I've expected to navigate within the textarea.  example: When there're multilines in TextField, and the cursor is in the last line. When I use the up arrow key, I expected the cursor to go to the previous line within the TextField, not jump to the previous row in DetailList.

### Priorities and help requested:

Are you willing to submit a PR to fix?  No

Requested priority: Normal

Products/sites affected: Unexpected user experience

**Comment by aneeshack4** [comment]
@woohling this doesn't repro for me - https://codepen.io/aneeshack4/pen/XWbejZq?editors=1010 I'm able to navigate and use arrow keys for focus just fine with multiline textfields.

**Comment by aneeshack4** [comment]
Acutally double checked to make sure this isn't a TextField problem - in multiline the up arrow moves the focus up to the previous line in of text so in FocusZone it should behave the same.

**Comment by aneeshack4** [comment]
I set breakpoints in FocusZone.tsx and stepping through them, it looks like `shouldInputLoseFocusOnArrowKey` doesn't get called in the case of the up and down arrows. Simply adding this call in the `KeyCodes` switch case for up and down doesn't fix this. I think there's something deeper to all this - there's a test case for 

> focus should leave input box when arrow keys are pressed when tabbing is supported but
    shouldInputLoseFocusOnArrowKey callback method return true

So perhaps this callback has to be defined with a return value of false....

**Comment by aneeshack4** [comment]
Updated the Codepen https://codepen.io/aneeshack4/pen/XWbejZq?editors=1011 to have tabbing explicitly supported (not none or undefined) `handleTabKey: FocusZoneTabbableElements.all` and passed in a callback that returns false for `shouldInputLoseFocusOnArrowKey` so that's also defined. According to this [comment](https://github.com/OfficeDev/office-ui-fabric-react/blob/master/packages/react-focus/src/components/FocusZone/FocusZone.tsx#L1161), that should mean the input doesn't lose focus... but it still is...

**Comment by woohling** [comment]
Hi @aneeshack4 , thanks for taking a look at this! 
The comment you are refering to  
`
private _shouldInputLoseFocus(element: HTMLInputElement, isForward?: boolean): boolean 
`
I suppose this function is handling input element? When multiple is set to `true`, the element is switched to textarea. Can this be the reason that it didn't work?

**Comment by woohling** [comment]
any progress on this issue?

**Comment by aneeshack4** [comment]
@woohling Apologies for the delay, I've been swamped with other issues. Yes I think you're onto it - you can see in `_moveFocus` here that if the element is an input (like `textarea`) it will return false - I believe that's the cause of it. I think rearchitecting this will take a significant amount of work so I'm going to add this to backlog. 

Unfortunately we don't currently have a codeowner for DetailsList but we plan to write a leaner, better version of it in the near future. So I don't know about the timeline yet but we'll publish a roadmap soon for Fluent overall (the next evolution of Fabric) that will make this clear.

**Comment by Dalimil** [comment]
@aneeshack4 @woohling I can confirm that this is an issue for not just the DetailList, but in fact for any multiline textarea inside a `FocusZone`. One cannot navigate within the textarea using arrows.
